functions.js

```diff
@@ -0,0 +1,61 @@
+const mysql = require('promise-mysql')
+const {set} = require("express/lib/application");
+const connection = mysql.createConnection({
+    host: '127.0.0.1',
+    user: 'root',
+    password: 'password',
+    database: 'tube_info'
+})
+const getTubes = () => async (request, response) => {
+    const db = await connection
+    const tubeStops =  await db.query('SELECT `name`, `code`, `line`, `id` FROM `tube_info` ORDER BY `line`, `id`;')
+    response.json(tubeStops)
+}
+
+const getRoute = () => async (request, response) => {
+    const from = request.query.from
+    const to = request.query.to
+
+    const fromCode = from.slice(-3)
+    const toCode = to.slice(-3)
+    const db = await connection
+
+    const findingLinesQuery = await db.query("SELECT code, line, COUNT(line) AS `station_count` FROM `tube_info` WHERE `code` = ? OR `code` = ? GROUP BY `line` HAVING `station_count` > 1",
+        [fromCode, toCode])
```

Comment:
Nice work 👍

Reply:
It could be simpler, though, I think this would also work:

"SELECT `line` FROM `tube_info` WHERE `code` = ? OR `code` = ? GROUP BY `line` HAVING `count(id)` > 1"

Reply:
(as you're only using `line` from the results)